spec/requests/users_controller_spec.rb

```diff
@@ -70,5 +70,33 @@
         expect(JSON.parse(response.body)['errors']).to be_present
       end
     end
+
+    context 'Role mapping' do
+      let(:user_creator) { instance_double(UserCreator) }
```

Comment:
I'm a bit skeptical about this and believe it may introduce complexity for you guys.
The reason behind that was me not wanting to call the `:new` original method because it will require a lot of mocking for the `SettingGetter` service.
So, I wanted a way to mock the `UserCreator` instance without doing any of that.
That appears to be the recommended way of doing it according to [RuboCop](https://www.rubydoc.info/gems/rubocop-rspec/1.7.0/RuboCop/Cop/RSpec/AnyInstance).

Reply:
Another problem was trying to assert the params given to the `UserCreator.new`.
The `user_params` is coming from the `user_controller#user_params` function that returns an instance of` `ActionController::Parameters` after requiring and permitting its values.

To avoid having to do some complex logic inside of the test `expect(user_creator).to receive(user_params: ActionController::Parameters.new(user_params).permit(...))` I decided to omit that assertion in favor of asserting the expected outcome by checking the `email` value to match what was given to the API.

Reply:
I think we could simplify this down to solve our problem. I would just test that the service is called properly and that's it. You are already testing the service in it's own file. 

Just expect that the params and provider are passed through

Reply:
There's no clean way that I can think of to make this as a unit test.
So, I'm changing it into a functional one.